lauth/spec/requests/authorized_any_spec.rb

```diff
@@ -0,0 +1,70 @@
+RSpec.describe "/authorized by username or client-ip", type: [:request, :database] do
+  let!(:collection) { Factory[:collection, :restricted_by_username_or_client_ip] }
+
+  # Setup networks, institution, and grant the instutition access to the collection
+  before(:each) do
+    institution = Factory[:institution]
+    Factory[ :network, :for_institution, institution: institution,
+      dlpsAccessSwitch: "allow", dlpsCIDRAddress: "10.1.16.0/24" ]
+    Factory[ :network, :for_institution, institution: institution,
+      dlpsAccessSwitch: "deny", dlpsCIDRAddress: "10.1.17.0/24" ]
+    Factory[:grant, :for_institution, institution: institution, collection: collection]
+  end
+
+  context "with an authorized individual" do
+    let!(:user) { Factory[:user, userid: "lauth-allowed"] }
+    let!(:user_grant) { Factory[:grant, :for_user, user: user, collection: collection] }
+
+    it "is allowed within an allowed network" do
+      expect(request(from: "10.1.16.1", as: user)).to eq({determination: "allowed"})
+    end
+    it "is allowed within a denied network" do
+      expect(request(from: "10.1.17.1", as: user)).to eq({determination: "allowed"})
+    end
+    it "is allowed outside of any known network" do
+      expect(request(from: "10.1.18.1", as: user)).to eq({determination: "allowed"})
+    end
+  end
+
+  context "with an authorized group" do
+    let!(:user) { Factory[:user, userid: "lauth-group-member"] }
+    let!(:group) {
+      Factory[:group]
+      relations.groups.last
+    }
+    let!(:group_membership) { Factory[:group_membership, group: group, user: user] }
+    let!(:group_grant) { Factory[:grant, :for_group, group: group, collection: collection] }
+
+    it "is allowed within an allowed network" do
+      expect(request(from: "10.1.16.1", as: user)).to eq({determination: "allowed"})
+    end
+    it "is allowed within a denied network" do
+      expect(request(from: "10.1.17.1", as: user)).to eq({determination: "allowed"})
+    end
+    it "is allowed outside of any known network" do
+      expect(request(from: "10.1.18.1", as: user)).to eq({determination: "allowed"})
+    end
+  end
```

Comment:
Can delete because this file is about username or ip, rather than all of the variations of how a username is granted.